test/services/MeasureService.test.ts

```diff
@@ -0,0 +1,44 @@
+import { initialize, Server } from '@projecttacoma/node-fhir-server-core';
+import { serverConfig } from '../../src/config/serverConfig';
+import { cleanUpDb, testSetup } from '../utils';
+import supertest from 'supertest';
+
+let server: Server;
+
+const Measure: fhir4.Measure = { resourceType: 'Measure', id: 'measure123', status: 'active' };
+
+describe('MeasureService', () => {
+  beforeAll(async () => {
+    const config = serverConfig;
+    server = initialize(config);
+
+    await testSetup([Measure]);
+  });
+
+  describe('searchById', () => {
+    it('test searchById with correctHeaders and the id should be in database returns 200', async () => {
+      await supertest(server.app)
+        .get('/4_0_1/Measure/measure123')
+        .set('Accept', 'application/json+fhir')
+        .expect(200)
+        .then(response => {
+          expect(response.body.id).toEqual(Measure.id);
+        });
+    });
+
+    it('test searchById when the id cannot be found in the database', async () => {
+      await supertest(server.app)
+        .get('/4_0_1/Measure/invalidID')
+        .set('Accept', 'application/json+fhir')
+        .expect(404)
+        .then(response => {
+          expect(response.body.issue[0].code).toEqual('ResourceNotFound');
+          expect(response.body.issue[0].details.text).toEqual(
+            `No resource found in collection: Measure, with: id invalidID`
+          );
+        });
+    });
+  });
+
+  afterAll(cleanUpDb);
```

Comment:
I was curious about the usage of afterAll here and how a reference to the function is passed in.

I'm not actually sure if this makes a difference at all, but I was looking at the [Jest docs for setup and teardown](https://jestjs.io/docs/setup-teardown#one-time-setup) and it seems like the example there is to return the actual promise from these hooks when the operations are asynchronous. Maybe we could change it to this approach? It could potentially help us avoid database race condition issues in tests that we've seen before in deqm-test-server, idk. This would make it look like this (and would apply to LibraryService.test.ts as well:


```ts
  beforeAll(() => {
    const config = serverConfig;
    server = initialize(config);
    return testSetup([Library]);
  });

// ...

  afterAll(() => {
    return cleanUpDb();
  });

```